rpm/lua-rex.spec

```diff
@@ -0,0 +1,243 @@
+Name:           lrexlib
+Version:        2.9.0
+Release:        1%{?dist}
+Summary:        Regular expression handling library for Lua
+
+Group:          Development/Libraries
+License:        MIT
+URL:            http://lrexlib.luaforge.net/
+Source0:        lrexlib-%{version}.tar.gz
+
+%description
+Lrexlib are bindings of six regular expression library APIs (POSIX, PCRE, PCRE2,
+Oniguruma, Tre and GNU) to Lua.
+
+%package pcre
+Summary: Lua binding of PCRE library
+BuildRequires:  pcre-devel
+BuildRequires:  pkgconfig
+BuildRequires:  tarantool-devel >= 1.7.7.0
+BuildRequires:	luarocks
+Requires:       tarantool >= 1.9.0.0
+Provides:       lrexlib-pcre = %{version}
+
+%description pcre
+Binding of PCRE library
+
+%package pcre2
+Summary: Lua binding of PCRE-2 library
+BuildRequires:  pcre2
+BuildRequires:  pcre2-devel
+BuildRequires:  pkgconfig
+BuildRequires:  tarantool-devel >= 1.7.7.0
+BuildRequires:	luarocks
+Requires:       tarantool >= 1.9.0.0
+Provides:       lrexlib-pcre2 = %{version}
+
+%description pcre2
+Binding of PCRE2 library
+
+%package posix
+Summary: Lua binding of POSIX library
+BuildRequires:  pkgconfig
+BuildRequires:  tarantool-devel >= 1.7.7.0
+BuildRequires:	luarocks
+Requires:       tarantool >= 1.9.0.0
+Provides:       lrexlib-posix = %{version}
+
+%description posix
+Binding of POSIX library
+
+%package gnu
+Summary: Lua binding of GNU library
+BuildRequires:  pkgconfig
+BuildRequires:  tarantool-devel >= 1.7.7.0
+BuildRequires:	luarocks
+Requires:       tarantool >= 1.9.0.0
+Provides:       lrexlib-gnu = %{version}
+
+%description gnu
+Binding of GNU library
+
+%package oniguruma
+Summary: Lua binding of Oniguruma library
+BuildRequires:  oniguruma-devel
+BuildRequires:  pkgconfig
+BuildRequires:  tarantool-devel >= 1.7.7.0
+BuildRequires:  luarocks
+Requires:       tarantool >= 1.9.0.0
+Provides:       lrexlib-oniguruma = %{version}
+
+%description oniguruma
+Binding of Oniguruma library
+
+#%package tre
+#Summary: Lua binding of TRE library
+#BuildRequires:  tre-devel >= 0.8.0
+#BuildRequires:  tre >= 0.8.0
+#BuildRequires:  pkgconfig
+#BuildRequires:  tarantool-devel >= 1.7.7.0
+#BuildRequires:  luarocks
+#Requires:       tarantool >= 1.9.0.0
+#Provides:       lrexlib-tre = %{version}
+
+#%description tre
+#Binding of TRE library
+
+%prep
+%setup -q -n lrexlib-%{version}
+
+for i in pcre pcre2 posix oniguruma gnu; do
+    curl --remote-name http://luarocks.org/manifests/rrt/lrexlib-$i-2.9.0-1.src.rock
+done
+
+%build
+mkdir tree
+for i in pcre pcre2 posix oniguruma gnu; do
+	TMP=$PWD/tmp luarocks --local --tree=./tree build lrexlib-$i-2.9.0-1.src.rock \
+	CFLAGS="%{optflags} -fPIC -DLUA_COMPAT_APIINTCASTS -I/usr/include/tarantool" \
+	PCRE_LIBDIR=%{_libdir} PCRE2_LIBDIR=%{_libdir} ONIG_LIBDIR=%{_libdir} \
+	TRE_LIBDIR=%{_libdir} TRE_INCDIR=/usr/include
+done
+
+%install
+install -d %{buildroot}%{_datadir}
+
+LUA_VER=5.1
+
+%if 0%{?rhel}
+    LUA_VER=5.1
+%endif
+
+%if 0%{?fedora}
+    LUA_VER=5.3
```

Comment:
Are you intend to compile it for tarantool or not?

Reply:
I compile it against tarantool.
```
TMP=$PWD/tmp luarocks --local --tree=./tree build lrexlib-$i-2.9.0-1.src.rock \
CFLAGS="%{optflags} -fPIC -DLUA_COMPAT_APIINTCASTS -I/usr/include/tarantool"
```

Reply:
There is a little quirk. Luarocks uses lua(abi) and name directories with built libraries using version of lua(abi). In different OS luarocks package ask for different versions of lua(abi), so we need the conditional mentioned.
We could build our own luarocks package or don't use luarocks for build purposes at all (remove it with cmake or even make as C code we need to build has simple structure)

Reply:
BTW I changed -I/usr/include/tarantool to -I%{_prefix}/include/tarantool